api/tests/unit/integrations/lead_tracking/hubspot/test_unit_hubspot_lead_tracking.py

```diff
@@ -1,469 +1,348 @@
-    mock_get_contact.assert_called_once_with(user)
+    mock_client_existing_contact.create_lead_form.assert_not_called()
+    mock_client_existing_contact.get_contact.assert_called_once_with(user)
 
 
-def test_hubspot_with_new_contact_and_existing_organisation(
+def test_create_organisation_lead_creates_contact_when_not_found(
     organisation: Organisation,
-    settings: SettingsWrapper,
     mocker: MockerFixture,
 ) -> None:
     # Given
-    settings.ENABLE_HUBSPOT_LEAD_TRACKING = True
     user = FFAdminUser.objects.create(
         email="new.user@example.com",
         first_name="Frank",
         last_name="Louis",
         marketing_consent_given=True,
     )
-    hubspot_id = "10280696017"
-
-    # Create an existing hubspot organisation to mimic a previous
-    # successful API call with a different lead.
-    HubspotOrganisation.objects.create(organisation=organisation, hubspot_id=hubspot_id)
-
-    mock_create_company = mocker.patch(
-        "integrations.lead_tracking.hubspot.client.HubspotClient.create_company"
-    )
-    mock_get_contact = mocker.patch(
-        "integrations.lead_tracking.hubspot.client.HubspotClient.get_contact",
-        return_value=None,
-    )
 
-    hubspot_lead_id = "1000551"
-    mock_create_contact = mocker.patch(
-        "integrations.lead_tracking.hubspot.client.HubspotClient.create_contact",
-        return_value={
-            "archived": False,
-            "archived_at": None,
-            "created_at": datetime.datetime(2024, 2, 26, 20, 2, 50, 69000),
-            "id": hubspot_lead_id,
-            "properties": {
-                "createdate": "2024-02-26T20:02:50.069Z",
-                "email": user.email,
-                "firstname": user.first_name,
-                "hs_all_contact_vids": "1000551",
-                "hs_email_domain": "example.com",
-                "hs_is_contact": "true",
-                "hs_is_unworked": "true",
-                "hs_marketable_status": user.marketing_consent_given,
-                "hs_object_id": "1000551",
-                "hs_object_source": "INTEGRATION",
-                "hs_object_source_id": "2902325",
-                "hs_object_source_label": "INTEGRATION",
-                "hs_pipeline": "contacts-lifecycle-pipeline",
-                "lastmodifieddate": "2024-02-26T20:02:50.069Z",
-                "lastname": user.last_name,
-            },
-            "properties_with_history": None,
-            "updated_at": datetime.datetime(2024, 2, 26, 20, 2, 50, 69000),
-        },
+    mock_client = mocker.MagicMock()
+    mock_client.get_contact.side_effect = [None, {"id": HUBSPOT_USER_ID}]
+    mock_client.create_lead_form.return_value = {}
+    mock_client.create_company.return_value = {"id": HUBSPOT_COMPANY_ID}
+    mock_client.get_company_by_domain.return_value = None
+    mocker.patch(
+        "integrations.lead_tracking.hubspot.lead_tracker.HubspotLeadTracker._get_client",
+        return_value=mock_client,
     )
```

Comment:
```suggestion
    mocker.patch(
        "integrations.lead_tracking.hubspot.lead_tracker.HubspotLeadTracker.HubspotClient",
        return_value=mock_client,
    )
```

I think this should give us the same behaviour, but means we don't have to mock a private method.

Reply:
Better indeed, thanks